### PR TITLE
CEM-1016 Typescript Errors React-UI

### PR DESCRIPTION
**Pull request by Ampatte2**
Upgrading typescript version and parser caused default rules to be updated as well.

Updated eslintrc.js with rules.

### CODE REVIEW

**Review by ralph-dev — COMMENTED** [review]
LGTM

Does that need return undefined? I'm not sure void === undefined? For the function part though

File: src/Containers/ImageCarousel.tsx [hunk 1]
```diff
@@ -25,7 +25,9 @@ export interface ImageCarouselProps
 
 export const ImageCarousel: React.FC<ImageCarouselProps> = ({
     imageData,
-    onClick = (): void => {},
+    onClick = (): void => {
+        return undefined;
```
Comment (ralph-dev) [inline src/Containers/ImageCarousel.tsx:29]: Weird

File: src/Inputs/MaskedInput.tsx [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
 
 export enum MaskedInputPreset {
     DOLLAR = 'DOLLAR',
     PERCENTAGE = 'PERCENTAGE',
 }
 
-export interface MaskedInputProps
-    extends LabelLayoutProps,
-        React.InputHTMLAttributes<HTMLInputElement> {
+export interface MaskedInputProps extends LabelLayoutProps, InputFragmentProps {
```
Comment (ralph-dev) [inline src/Inputs/MaskedInput.tsx:14]: No need for HTML? There is a props spread, is that covered by the others?

**Review by ralph-dev — APPROVED** [review]
Lgtm

File: src/Inputs/MaskedInput.tsx (hunk 2 above)
Comment (Ampatte2) [reply src/Inputs/MaskedInput.tsx:14]: InputFragmentProps already extends HTMLInputElement.

File: src/Containers/ImageCarousel.tsx (hunk 1 above)
Comment (Ampatte2) [reply src/Containers/ImageCarousel.tsx:29]: Either turn off no empty functions, or this is the correct way to type a void function in typescript.

**Comment by Ampatte2** [comment]
Typing wise void === undefined.

File: src/Containers/ImageCarousel.tsx (hunk 1 above)
Comment (ralph-dev) [reply src/Containers/ImageCarousel.tsx:29]: Let's turn off no empty functions then @fpunny

**Comment by Ampatte2** [comment]
I do not know why but removing quiet fixes the checks and build issues.

**Review by ralph-dev — APPROVED** [review]
LGTM, Good catch!

**Pull request merged**